### ISSUE TITLE
Selecting a checkbox within a table cell

### ISSUE THREAD

**Issue by tawalters61** [angular/material2#8600]
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
When selecting a checkbox within a table, the highlighted selection highlighting should not move up.

#### What is the current behavior?
The alignment of the highlighting is not aligned with the checkbox itself.

#### What are the steps to reproduce?
Providing a StackBlitz/Plunker (or similar) is the *best* way to get the team to see your issue. <br/>
Plunker starter (using on `@master`): https://goo.gl/uDmqyY<br/>
StackBlitz starter (using latest `npm` release): https://goo.gl/wwnhMV<br/>

This existing table example does not work properly with material:

#### What is the use-case or motivation for changing an existing behavior?
It's a bug

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Material: 5.0.0-rc0, 5.0.0-rc.1
The bug does not exist in 2.0.0-beta.12

#### Is there anything else we should know?
No

**Comment by jelbourn** [comment]
@tawalters61 I don't follow what the issue is you're reporting.

**Comment by tawalters61** [comment]
When selecting a checkbox within a table cell, the checkbox moves up several pixels, causing it not to be aligned properly. Also, the 'special effects' are clipped. In the link above, it works correctly, because the example online is using a pre-5.0.0 version of Material . But, if the same example is run with Material 5.0.0-rc0 or 5.0.0-rc.1, the problem occurs.

**Comment by n-marchenko** [comment]
@jelbourn The checkbox is only moved when you directly click on it. If the ripple effect is not applied, then everything is fine

![checkbox_bug](https://user-images.githubusercontent.com/17332113/33158448-3dce5d3a-d034-11e7-83b4-97527beef3c9.gif)

**Comment by tawalters61** [comment]
Yes, this is what I am seeing. This problem does not exist in the pre-5 versions.

**Comment by wandri** [comment]
Same as this issue: https://github.com/angular/material2/issues/8517 .
One Workaround to wait the fix is:

```
  mat-cell {
    overflow: visible;
  }
```

**Comment by tawalters61** [comment]
Thanks! This worked!

**Comment by volei** [comment]
Actually, I remember having this effect since way before Material 5.0.0-rc0 on Windows7/Chrome/Firefox/Typescript 2.4.x.

**Comment by merschel** [comment]
Could be related to #8517 and #8588.

**Comment by andrewseguin** [comment]
The issue indeed is due to overflow. For some reason the checkbox doesn't like how the cell's overflow is set and it moves when clicked. Need to investigate why the checkbox starts correctly but the moves.

**Comment by pacurtin** [comment]
// Bug fix for mat-checkbox inside a mat-cell
mat-cell, .mat-header-cell{
   overflow: visible;
 }

If you want to put a checkbox in the header (i.e "Select All" checkbox) add .mat-header-cell to the CSS as well.

**Comment by tonysamperi** [comment]
Same in @angular/material@5.0.4

**Comment by DouglasWebster** [comment]
I have the same problem when using an unlabelled  `mat-checkbox `inside  `mat-cell`.  The problem disappeared when I gave it a label.  For the present I use `<mat-checkbox>&nbsp;</mat-checkbox>` as I find it quicker that altering the css  Of course, this only works if you can afford the extra space!

**Comment by tonysamperi** [comment]
@DouglasWebster Good to know! Actually I could have afforded to "steal" extra space...I'll give it a try with the next table ;)

**Comment by arundhaj** [comment]
@wandri @pacurtin @tonysamperi solution fixes the checkbox issue, however it mess up other columns alignment

**Comment by tonysamperi** [comment]
@arundhaj no it doesn't. Tomorrow I'll post a screen.
Probably you have a different environment...

**Comment by JuanUicab** [comment]
Hello,
Just add a space or "Alt + 0160" like this:

<mat-checkbox> </mat-checkbox>

It works for me.

**Comment by lostllama** [comment]
Any ETA for a non-hacky fix for this?

**Comment by tbunique** [comment]
I have this problem with latest version of material.
None of the hacks worked, or should I say, they sometimes work, sometimes not. It's pretty random.

I'm trying to find a workaround that will work in my situation.

**Comment by akvaliya** [comment]
I have checkbox at the bottom of page & in mobile devices when click on it moves scrolls & hides below. So we are unable to check/uncheck in mobile devices.

![ezgif-4-dc9dc44ceaad](https://user-images.githubusercontent.com/25152577/66647273-17a3fe00-ec46-11e9-90ac-9f82f92edc92.gif)


Any resolution for this?

**Comment by abha01singh** [comment]
> I have checkbox at the bottom of page & in mobile devices when click on it moves scrolls & hides below. So we are unable to check/uncheck in mobile devices.
> 
> ![ezgif-4-dc9dc44ceaad](https://user-images.githubusercontent.com/25152577/66647273-17a3fe00-ec46-11e9-90ac-9f82f92edc92.gif)
> 
> Any resolution for this?

I am facing the same issue.
looks like mat-option does not support float property and gives this kind of behaviour to the form.

Any suggestion for an alternate of float ??

**Comment by benoit-laplante** [comment]
I did this as a workaround using ng11:
`:host ::ng-deep mat-cell mat-checkbox span {
  overflow: unset;
}`

**Comment by bugoutfr** [comment]
Hello, is there any plan to fix this bug? It's been open since 2017!
The suggested workarounds didn't work for me.
